src/platform/qemu-aarch64-virt/inc/plat/psci.h

```diff
@@ -9,8 +9,9 @@
 #define PSCI_POWER_STATE_LVL_0    0x0000000
 #define PSCI_POWER_STATE_LVL_1    0x1000000
 #define PSCI_POWER_STATE_LVL_2    0x2000000
-#define PSCI_STATE_TYPE_STANDBY   0x00000
+#define PSCI_STATE_TYPE_STANDBY   0x1
 #define PSCI_STATE_TYPE_BIT       (1UL << 16)
-#define PSCI_STATE_TYPE_POWERDOWN (0x1)
+#define PSCI_STATE_ID_RETENTION   0x2
+#define PSCI_STATE_TYPE_POWERDOWN PSCI_STATE_TYPE_BIT | PSCI_STATE_ID_RETENTION
```

Comment:
Shouldn't powerdown be non-retentive?

Reply:
I believe there may be some confusion with the power states on RISC-V. In RISC-V, platform-specific suspend states for a hart can be either retentive or non-retentive: a retentive suspend state preserves hart registers and CSR values across all privilege modes, while a non-retentive state does not. 
However, the terminology and structure differ on ARM, where the nomenclature for power states is not the same.

An ARM core can exist in one of four power states:
- Run: The core is fully powered and operational
- Standby: The core remains powered, but energy-saving measures are applied (`wfi`), preserving all core states (this power state is similar to RISC-V retention state)
- Retention: The core is partially powered down, with its state saved and restored upon resuming normal operation (this power state is similar to RISC-V non-retention state)
- Powerdown: The core is fully powered off.

More details:
The ARM `CPU_SUSPEND` API enables a topology node to enter a low-power state. 
The `power_state` bit 16 in the CPU suspend call specifies the state type and bits [15:0] the State ID. 
For power-down, bit 16 is set to 1, and the State ID represents a platform-specific state. 
In [ATF’s QEMU implementation](https://github.com/bao-project/arm-trusted-firmware/blob/bao/master/plat/qemu/common/qemu_pm.c#L54), three idle states are defined: Standby (StateId = 0x1) and two power-down states — one for the core level and one for the cluster level — both categorized as retention states (StateId = 0x2)

Reply:
Ok! That makes sense!